extensions/mssql/src/objectManagement/ui/restoreDatabaseDialog.ts

```diff
@@ -231,34 +230,52 @@ export class RestoreDatabaseDialog extends ObjectManagementDialogBase<Database,
 		this.restoreFrom = this.createDropdown(localizedConstants.RestoreFromText, async (newValue) => {
 			if (newValue === localizedConstants.RestoreFromBackupFileOptionText || newValue === localizedConstants.RestoreFromUrlText) {
 				this.backupFilePathContainer.display = 'inline-flex';
 				this.restoreDatabase.enabled = false;
-				this.backupFilePathInput.value = newValue === localizedConstants.RestoreFromUrlText ? this.backupURLPath : this.backupFilePath;
+				this.backupFilePathInput.enabled = false;
+				await this.backupFilePathInput.updateProperties({
+					value: newValue === localizedConstants.RestoreFromUrlText ? this.backupURLPath : this.backupFilePath
+				});
 			} else if (newValue === localizedConstants.RestoreFromDatabaseOptionText) {
 				this.backupFilePathContainer.display = 'none';
 				this.restoreDatabase.enabled = true;
 			}
+			if (newValue === localizedConstants.RestoreFromS3UrlText) {
+				this.backupFilePathInput.enabled = true;
+				this.backupFilePathContainer.display = 'inline-flex';
+				this.restoreDatabase.enabled = false;
+				const credentials = (await this.objectManagementService.getCredentialNames(this.options.connectionUri)).filter(url => url.startsWith('s3://'));
+				credentials.unshift('');
+				await this.backupFilePathInput.updateProperties({
+					value: '',
+					values: credentials
+				});
+			}
 			await this.updateNewRestorePlanToDialog();
 		}, restoreFromDropdownOptions, restoreFromDropdownOptions[0], !this.isManagedInstance, RestoreInputsWidth);
 		containers.push(this.createLabelInputContainer(localizedConstants.RestoreFromText, this.restoreFrom));
 
 		// Backup file path
-		this.backupFilePathInput = this.createInputBox(async (newValue) => {
+		this.backupFilePathInput = this.createDropdown(localizedConstants.BackupFilePathText, async (newValue) => {
 			await this.updateNewRestorePlanToDialog();
-		}, {
-			ariaLabel: localizedConstants.BackupFilePathText,
-			inputType: 'text',
-			enabled: true,
-			value: '',
-			width: RestoreInputsWidth - 30,
-			placeHolder: localizedConstants.BackupFolderPathTitle
-		});
+		}, [''], localizedConstants.BackupFolderPathTitle, false, RestoreInputsWidth - 30, false, true)
+
 		this.backupFilePathButton = this.createButton('...', '...', async () => {
-			this.restoreFrom.value === localizedConstants.RestoreFromUrlText
-				? await this.createBackupUrlFileBrowser() : await this.createBackupFileBrowser()
+			switch (this.restoreFrom.value) {
+				case localizedConstants.RestoreFromUrlText:
+					await this.createBackupUrlFileBrowser();
+					break;
+				case localizedConstants.RestoreFromBackupFileOptionText:
+					await this.createBackupFileBrowser();
+					break;
+				case localizedConstants.RestoreFromS3UrlText:
```

Comment:
I'm not sure re-using this button is the best option here - I would expect this to actually open a file browser dialog instead of just creating a credential. 

Instead, what about having a "Generate credential" button that appears below the dropdown. When the user clicks it it'll pop up the dialog and then switches to have that selected once it's done creating the new credential.

Although there's even more updates you could make as well. For example - if I'm understanding the docs correctly the user can put whatever URL they want to and the expectation is that there's a matching credential in the format `s3://<endpoint>:<port>/<bucket>` that's what it'll use. So you could even have this dialog do a check to see if such a credential exists and select it if so, otherwise prompt the user to create a new credential.